vision/google/cloud/vision/image.py

```diff
@@ -85,28 +85,58 @@ def source(self):
         """
         return self._source
 
-    def _detect_annotation(self, feature):
+    def _detect_annotation(self, features):
         """Generic method for detecting a single annotation.
 
-        :type feature: :class:`~google.cloud.vision.feature.Feature`
-        :param feature: The ``Feature`` indication the type of annotation to
-                        perform.
+        :type features: list of:class:`~google.cloud.vision.feature.Feature`
+        :param features: List of the ``Feature``s indicating the type of
+                         annotations to perform.
 
         :rtype: list
         :returns: List of
-                  :class:`~google.cloud.vision.entity.EntityAnnotation`.
+                  :class:`~google.cloud.vision.entity.EntityAnnotation`,
+                  :class:`~google.cloud.vision.face.Face`,
+                  :class:`~google.cloud.vision.color.ImagePropertiesAnnotation`,
+                  :class:`~google.cloud.vision.sage.SafeSearchAnnotation`,
         """
+        detected_objects = []
+        results = self.client.annotate(self, features)
+        for feature in features:
+            detected_objects.extend(
+                self._entity_from_response_type(feature.feature_type, results))
+
+        if len(detected_objects) == 1:
+            return detected_objects[0]
+        return detected_objects
+
+    @staticmethod
+    def _entity_from_response_type(feature_type, results):
+        """Convert a JSON result to an entity type based on the feature."""
+
         reverse_types = {
+            'FACE_DETECTION': 'faceAnnotations',
+            'IMAGE_PROPERTIES': 'imagePropertiesAnnotation',
             'LABEL_DETECTION': 'labelAnnotations',
             'LANDMARK_DETECTION': 'landmarkAnnotations',
             'LOGO_DETECTION': 'logoAnnotations',
+            'SAFE_SEARCH_DETECTION': 'safeSearchAnnotation',
             'TEXT_DETECTION': 'textAnnotations',
         }
         detected_objects = []
-        result = self.client.annotate(self, [feature])
-        for response in result[reverse_types[feature.feature_type]]:
-            detected_object = EntityAnnotation.from_api_repr(response)
-            detected_objects.append(detected_object)
+        feature_key = reverse_types[feature_type]
+
+        if feature_type == 'FACE_DETECTION':
+            for face in results[feature_key]:
+                detected_objects.append(Face.from_api_repr(face))
+        elif feature_type == 'IMAGE_PROPERTIES':
+            detected_objects.append(
+                ImagePropertiesAnnotation.from_api_repr(results[feature_key]))
+        elif feature_type == 'SAFE_SEARCH_DETECTION':
+            result = results[feature_key]
+            detected_objects.append(SafeSearchAnnotation.from_api_repr(result))
+        else:
+            for result in results[feature_key]:
```

Comment:
Yeah, `LABEL_DETECTION`, `LANDMARK_DETECTION`, `LOGO_DETECTION` and `TEXT_DETECTION` all return `EntityAnnotation`s.